routes/extra_data_encoding.go

```diff
@@ -0,0 +1,200 @@
+package routes
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"github.com/deso-protocol/core/lib"
+	"github.com/golang/glog"
+	"github.com/pkg/errors"
+	"strconv"
+)
+
+type ExtraDataDecoderFunc func([]byte, *lib.DeSoParams, *lib.UtxoView) string
+type ExtraDataEncoderFunc func(string) ([]byte, error)
+
+type ExtraDataEncoding struct {
+	Decode ExtraDataDecoderFunc
+	Encode ExtraDataEncoderFunc
+}
+
+// specialExtraDataKeysToEncoding These are reserved extra data fields used in core that have special encoding / decoding.
+// These fields are populated directly in core, but we still want to allow clients to be able to populate them directly
```

Comment:
Alternatively, we can use `ReservedFieldCannotOverride` to reject creation for any transaction where an ExtraData field populated by a client collides with one of these reserved fields used in core.

Reply:
I think it's okay to leave it flexible for now. The transactions will be rejected if the extra data has a reserved key that doesn't contain a legitimate value.